src/simularium/SelectionInterface.ts

```diff
@@ -276,19 +271,39 @@ class SelectionInterface {
     }
 
     private updateUiDataColor(
-        entry: UIDisplayEntry,
-        id: number,
+        agentName: string,
+        idsToUpdate: number[],
         color: number | string
-    ) {
-        const tagsToUpdate = this.getTags(entry.name, id);
-        entry.displayStates.forEach((displayState: DisplayStateEntry) => {
-            if (tagsToUpdate.includes(displayState.id)) {
-                displayState.color = convertColorNumberToString(color);
+    ): void {
+        const newColor = convertColorNumberToString(color);
+        const entry = this.entries[agentName];
+        entry.forEach((displayState) => {
+            if (idsToUpdate.includes(displayState.id)) {
+                displayState.color = newColor;
+            }
+        });
+    }
```

Comment:
This might necessitate a change on the front end. As it was, it was modified the uiDisplay data, which is derived from the selectionInterface. But it's not being stored. Now when you adjust the color it's being stored in the selection interface data